src/rust/bin/wallet_tool.rs

```diff
@@ -0,0 +1,580 @@
+    exec(&zcash_cli, &cli_args, false)
+        .and_then(|out| {
+            let cli_err: Vec<_> = from_utf8(&out.stderr)
+                .with_context(|| "Output from zcash-cli was not UTF-8")?
+                .lines()
+                .map(|s| s.trim_end_matches('\r'))
+                .collect();
+            if opts.debug {
+                eprintln!("DEBUG: stderr {:?}", cli_err);
+            }
+            if !cli_err.is_empty() {
+                if cli_err[0].starts_with("error: couldn't connect") {
+                    println!("\nWe could not connect to zcashd; it may have exited.");
+                    return Err(WalletToolError::ZcashdConnection.into());
+                } else {
+                    println!(
+                        "\nThere was an unexpected response from zcashd:\n> {}",
+                        cli_err.join("\n> "),
+                    );
+                    return Err(WalletToolError::UnexpectedResponse.into());
+                }
+            }
+            println!(concat!(
+                "\nThe backup of the emergency recovery phrase for the zcashd\n",
+                "wallet has been successfully confirmed 🙂. You can now use the\n",
+                "zcashd RPC methods that create keys and addresses in that wallet.\n\n",
+                "If you use other wallets, their recovery information will need\n",
+                "to be backed up separately.\n"
+            ));
+            Ok(())
+        })
+        .map_err(|e| {
+            println!(concat!(
+                "\nzcash-wallet-tool was unable to communicate to zcashd that the\n",
+                "backup was confirmed. This can happen if zcashd stopped, in which\n",
+                "case you should try again. If zcashd is still running, please seek\n",
+                "help or try to use 'zcash-cli -stdin walletconfirmbackup' manually.\n"
+            ));
+            e
+        })?;
+    Ok(())
+}
+
+fn prompt<'a>(input: &mut Stdin, buf: &'a mut String) -> anyhow::Result<&'a str> {
+    input
+        .read_line(buf)
+        .with_context(|| "Error reading from stdin")?;
+
+    if !buf.ends_with('\n') {
+        Err(io::Error::from(ErrorKind::UnexpectedEof))
+            .with_context(|| "End of file reading from stdin")
+    } else {
+        Ok(buf.trim_end_matches(|c| c == '\r' || c == '\n').trim())
+    }
+}
+
+fn ordinal(num: usize) -> String {
+    let suffix = if (11..=13).contains(&(num % 100)) {
+        "th"
+    } else {
+        match num % 10 {
+            1 => "st",
+            2 => "nd",
+            3 => "rd",
+            _ => "th",
+        }
+    };
+    format!("{}{}", num, suffix)
+}
+
+fn zcash_cli_path(debug: bool) -> anyhow::Result<PathBuf> {
+    // First look for `zcash_cli[.exe]` as a sibling of the executable.
+    let mut exe = env::current_exe()
+        .with_context(|| "Cannot determine the path of the running executable")?;
+    exe.set_file_name("zcash-cli");
+    exe.set_extension(EXE_EXTENSION);
```

Comment:
Is this necessary? When would the extension of the current executable not be correct?

EDIT: I agree we need to do this further down, so I'm not against doing this here for similarity.

Reply:
`set_file_name` clears the extension. (The implementation does a `.pop()` first and then pushes `file_name` without an extension.)